### ISSUE TITLE
Ruboto setup fails: get_android_sdk_version

### ISSUE THREAD

**Issue by davenguyen** [ruboto/ruboto#806]
Mac OS X (El Capitan 10.11.5)
ruboto (1.4.1)
ruby (tested on 2.3.1 and 2.2.3)

I just downloaded the ruboto gem and getting a problem when running `ruboto setup -y`:

``` shell
Java runtime             : Found
Java Compiler            : Found
Apache ANT               : Not found
Android Package Installer: Not found
Android Emulator         : Not found
Intel HAXM               : Not found
Android SDK Command adb  : Not found
Android SDK Command dx   : Not found
Platform SDK android-16  : Not found

    !!! Ruboto setup is NOT OK !!!

Android package installer not found.
F, [2016-07-03T06:29:45.369101 #84287] FATAL -- : undefined method `[]' for nil:NilClass (NoMethodError)
/Users/dave/.rvm/gems/ruby-2.3.1@ruboto/gems/ruboto-1.4.1/lib/ruboto/util/setup.rb:153:in `get_android_sdk_version'
/Users/dave/.rvm/gems/ruby-2.3.1@ruboto/gems/ruboto-1.4.1/lib/ruboto/util/setup.rb:528:in `block in install_android_sdk'
/Users/dave/.rvm/gems/ruby-2.3.1@ruboto/gems/ruboto-1.4.1/lib/ruboto/util/setup.rb:525:in `chdir'
/Users/dave/.rvm/gems/ruby-2.3.1@ruboto/gems/ruboto-1.4.1/lib/ruboto/util/setup.rb:525:in `install_android_sdk'
/Users/dave/.rvm/gems/ruby-2.3.1@ruboto/gems/ruboto-1.4.1/lib/ruboto/util/setup.rb:306:in `install_all'
/Users/dave/.rvm/gems/ruby-2.3.1@ruboto/gems/ruboto-1.4.1/lib/ruboto/util/setup.rb:24:in `setup_ruboto'
/Users/dave/.rvm/gems/ruby-2.3.1@ruboto/gems/ruboto-1.4.1/lib/ruboto/commands/base.rb:467:in `run'
/Users/dave/.rvm/gems/ruby-2.3.1@ruboto/gems/main-5.3.2/lib/main/program/class_methods.rb:155:in `block in run'
/Users/dave/.rvm/gems/ruby-2.3.1@ruboto/gems/main-5.3.2/lib/main/program/class_methods.rb:144:in `catch'
/Users/dave/.rvm/gems/ruby-2.3.1@ruboto/gems/main-5.3.2/lib/main/program/class_methods.rb:144:in `run'
/Users/dave/.rvm/gems/ruby-2.3.1@ruboto/gems/main-5.3.2/lib/main/factories.rb:18:in `run'
/Users/dave/.rvm/gems/ruby-2.3.1@ruboto/gems/main-5.3.2/lib/main/factories.rb:25:in `Main'
/Users/dave/.rvm/gems/ruby-2.3.1@ruboto/gems/ruboto-1.4.1/lib/ruboto/commands/base.rb:28:in `main'
/Users/dave/.rvm/gems/ruby-2.3.1@ruboto/gems/ruboto-1.4.1/bin/ruboto:11:in `<top (required)>'
/Users/dave/.rvm/gems/ruby-2.3.1@ruboto/bin/ruboto:23:in `load'
/Users/dave/.rvm/gems/ruby-2.3.1@ruboto/bin/ruboto:23:in `<main>'
/Users/dave/.rvm/gems/ruby-2.3.1@ruboto/bin/ruby_executable_hooks:15:in `eval'
/Users/dave/.rvm/gems/ruby-2.3.1@ruboto/bin/ruby_executable_hooks:15:in `<main>'
```

I dug into `lib/ruboto/util/setup.rb#get_android_sdk_version` and found that `page_content` returns blank. I checked this using these two lines in a ruby console:

```
2.3.1 :001 > SDK_DOWNLOAD_PAGE = 'http://developer.android.com/sdk/index.html?hl=sk'
 => "http://developer.android.com/sdk/index.html?hl=sk"
2.3.1 :002 > page_content = Net::HTTP.get(URI.parse(SDK_DOWNLOAD_PAGE))
 => ""
```

The link itself still works, and using a different URL does bring in page content, so it's just that link blocking the Net::HTTP.get call. If I manually return the version (`24.4.1`), it then downloads.

There are two missing packages afterwards:

``` shell
Java runtime             : Found
Java Compiler            : Found
Apache ANT               : Not found
Android Package Installer: Found
Android Emulator         : Found
Intel HAXM               : Found
Android SDK Command adb  : Found
Android SDK Command dx   : Not found
Platform SDK android-16  : Found

    !!! Ruboto setup is NOT OK !!!

Android tools not found.
Refresh Sources:
  Fetching https://dl.google.com/android/repository/addons_list-2.xml
  Validate XML
  Parse XML
  Fetched Add-ons List successfully
  Refresh Sources
  Fetching URL: https://dl.google.com/android/repository/repository-11.xml
  Validate XML: https://dl.google.com/android/repository/repository-11.xml
  Parse XML:    https://dl.google.com/android/repository/repository-11.xml
  Fetching URL: https://dl.google.com/android/repository/addon.xml
  Validate XML: https://dl.google.com/android/repository/addon.xml
  Parse XML:    https://dl.google.com/android/repository/addon.xml
  Fetching URL: https://dl.google.com/android/repository/glass/addon.xml
  Validate XML: https://dl.google.com/android/repository/glass/addon.xml
  Parse XML:    https://dl.google.com/android/repository/glass/addon.xml
  Fetching URL: https://dl.google.com/android/repository/extras/intel/addon.xml
  Validate XML: https://dl.google.com/android/repository/extras/intel/addon.xml
  Parse XML:    https://dl.google.com/android/repository/extras/intel/addon.xml
  Fetching URL: https://dl.google.com/android/repository/sys-img/android/sys-img.xml
  Validate XML: https://dl.google.com/android/repository/sys-img/android/sys-img.xml
  Parse XML:    https://dl.google.com/android/repository/sys-img/android/sys-img.xml
  Fetching URL: https://dl.google.com/android/repository/sys-img/android-wear/sys-img.xml
  Validate XML: https://dl.google.com/android/repository/sys-img/android-wear/sys-img.xml
  Parse XML:    https://dl.google.com/android/repository/sys-img/android-wear/sys-img.xml
  Fetching URL: https://dl.google.com/android/repository/sys-img/android-tv/sys-img.xml
  Validate XML: https://dl.google.com/android/repository/sys-img/android-tv/sys-img.xml
  Parse XML:    https://dl.google.com/android/repository/sys-img/android-tv/sys-img.xml
  Fetching URL: https://dl.google.com/android/repository/sys-img/google_apis/sys-img.xml
  Validate XML: https://dl.google.com/android/repository/sys-img/google_apis/sys-img.xml
  Parse XML:    https://dl.google.com/android/repository/sys-img/google_apis/sys-img.xml
Error: Ignoring unknown package filter 'build-tools-23.0.1'
Error: Ignoring unknown package filter 'extra-intel-hardware_accelerated_execution_manager'
Warning: The package filter removed all packages. There is nothing to install.
         Please consider trying to update again without a package filter.
Android SDK Command dx   : Not found
Android SDK Command adb  : Found
Android Emulator         : Found
Intel HAXM               : Found
Java runtime             : Found
Java Compiler            : Found
Apache ANT               : Not found
Android Package Installer: Found
Android Emulator         : Found
Intel HAXM               : Found
Android SDK Command adb  : Found
Android SDK Command dx   : Not found
Platform SDK android-16  : Found

    !!! Ruboto setup is NOT OK !!!
```

Not sure what to do from here.

**Comment by davenguyen** [comment]
I was able to get an AVD created. To install Apache Ant, I used `brew install ant`. To install dx, I ran `android update sdk` and selected "Android SDK Build-tools" from the options under Tools. Let me know if I can help resolve this issue the "proper" way.

**Comment by donv** [comment]
Hi @davenguyen !

Thanks for the report!  At first glance it looks like https://github.com/ruboto/ruboto/issues/715 , but that should have been fixed for ruboto 1.3.1.

Can you check out the Ruboto source and do a `rake install` to see if the current master works for you?

**Comment by donv** [comment]
Same as #811 and #824.  Should be fixed now.
